File: frontend/src/types/payment.ts
export type PaymentStatus =
  | "PENDING"
  | "COMPLETED"
  | "FAILED"
  | "CANCELLED"
  | "REFUNDED";

export type Payment = {
  id: string;
  booking_id: string;
  amount: number;
  currency: string;
  status: PaymentStatus;
  provider?: string | null;
  external_id?: string | null;
  paid_at?: string | null;
  created_at: string;
};

File: backend/app/schemas/payment.py
from datetime import datetime
from decimal import Decimal
from uuid import UUID
from pydantic import BaseModel
from enum import Enum


class PaymentStatus(str, Enum):
    PENDING = "PENDING"
    COMPLETED = "COMPLETED"
    FAILED = "FAILED"
    CANCELLED = "CANCELLED"
    REFUNDED = "REFUNDED"


class PaymentCreate(BaseModel):
    booking_id: UUID
    amount: Decimal
    currency: str = "VND"
    provider: str | None = None  # VNPAY, MOMO, etc.


class PaymentResponse(BaseModel):
    id: UUID
    booking_id: UUID
    amount: Decimal
    currency: str
    status: PaymentStatus
    provider: str | None = None
    external_id: str | None = None
    paid_at: datetime | None = None
    created_at: datetime

    model_config = {"from_attributes": True}
